### DISCUSSION TITLE
NEXT JS + twin macro theme

### DISCUSSION THREAD

**Discussion by wdcs-jayfatania** [ben-rogerson/twin.macro#782] [Q&A]
Hello @ben-rogerson, I am using twin macro in my next js project with typescript.

I am tryping to import twin macro theme to use colors of that but getting error like below,

TypeError: (0 , twin_macro__WEBPACK_IMPORTED_MODULE_1__.theme) is not a function

Here is my code,

```
import { theme } from 'twin.macro'

const twColor = theme`colors`

export const COLOR_1 = twColor.indigo['600'] 
export const COLOR_2 = twColor.blue['500']
export const COLOR_3 = twColor.emerald['500']
export const COLOR_4 = twColor.amber['500']
export const COLOR_5 = twColor.red['500']
export const COLOR_6 = twColor.purple['500']
export const COLOR_7 = twColor.cyan['500']

export const COLOR_1_LIGHT = twColor.indigo['100'] 
export const COLOR_2_LIGHT = twColor.blue['100']
export const COLOR_3_LIGHT = twColor.emerald['100']
export const COLOR_4_LIGHT = twColor.amber['100']
export const COLOR_5_LIGHT = twColor.red['100']
export const COLOR_6_LIGHT = twColor.purple['100']
export const COLOR_7_LIGHT = twColor.cyan['100']

```

I dont know what I am missing, can you help me out this please.

**Comment by ben-rogerson** [comment]
Hey there

Lately people are using a src/app directory with next so I think you may only need to adjust your project folders in the `withTwin.js` file:

```js
// withTwin.js
const includedDirs = [
  path.resolve(__dirname, 'src/components'),
  path.resolve(__dirname, 'src/pages'),
  path.resolve(__dirname, 'src/styles'),
]
```

Then make sure you're loading the file by wrapping your config:

```js
// next.config.js
const withTwin = require('./withTwin.js')

module.exports = withTwin({
    // ...
})
```

**Reply by wdcs-jayfatania** [reply]
Thanks @ben-rogerson, I have implement this way, but now its not even compiling the project, its stuck at the stage of compiling client and server, see the attached image.

![image](https://user-images.githubusercontent.com/88309705/218076363-341f5721-92bb-49f5-bb70-cc22888c708c.png)
